tensorflow_datasets/core/image_visualizer.py

```diff
@@ -0,0 +1,235 @@
+# coding=utf-8
+# Copyright 2020 The TensorFlow Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+
+"""Visualizer for image datasets.
+"""
+
+from absl import logging
+
+from tensorflow_datasets.core import dataset_utils
+from tensorflow_datasets.core import features as features_lib
+from tensorflow_datasets.core import lazy_imports_lib
+
+# Utility functions
+def _extract_keys(ds_info, feature_type, secondary_key=None):
+  """Extracts keys from ds_info based on feature type"""
+
+  if secondary_key:
+    return [
+        k for k, feature in ds_info.features[secondary_key].items()
+        if isinstance(feature, feature_type)
+    ]
+
+  return [
+      k for k, feature in ds_info.features.items()
+      if isinstance(feature, feature_type)
+  ]
+
+def _draw_text(ax, xy, txt, size=14):
+  """Draws text on the top-left corner of the bounding box"""
+  ax.text(*xy, txt, verticalalignment='top', \
+          color='white', fontsize=size, weight="normal", \
+          bbox=dict(facecolor='black', alpha=0.7, pad=0.3))
+
+def _draw_rectangle(ax, bbox):
+  """Draws the bounding box"""
+  patches = lazy_imports_lib.lazy_imports.matplotlib.patches
+  ax.add_patch(patches.Rectangle(bbox[:2], \
+              *bbox[-2:], fill=False, \
+              edgecolor='red', lw=0.7))
+
+def _bb_hw(bbox, height, width):
+  """converts bounding box to height and width"""
+  return [bbox[1]*width, bbox[0]*height, \
+          (bbox[3]*width)-(bbox[1]*width)+1, \
+          (bbox[2]*height)-(bbox[0]*height)+1]
+
+# Visualizer classes
+class ImageVisualizer():
+  """Parent Class for support of Image based datasets"""
+
+  def build(self, ds_info, ds, rows=3, cols=3, plot_scale=3, image_key=None):
+    """Builds the plot for visualization"""
+
+    self.plot_scale = plot_scale
+    image_key = self._infer_image_key(ds_info, image_key)
+    object_dict = self._infer_object_keys(ds_info)
+
+    num_examples = rows * cols
+    examples = list(dataset_utils.as_numpy(ds.take(num_examples)))
+
+    plt = lazy_imports_lib.lazy_imports.matplotlib.pyplot
+    fig = plt.figure(figsize=(plot_scale*cols, plot_scale*rows))
+    fig.subplots_adjust(hspace=1/plot_scale, wspace=1/plot_scale)
+
+    for i, ex in enumerate(examples):
+      ax = self._plot_image(plt, fig, ex, rows, cols, i, image_key)
```

Comment:
As a general rules, when the number of argument becomes too big, it's best to use keywords arguments:
```
_plot_image(
    plt=plt,
    ex=ex,
    ...
)
```
This make the code less error prone an more maintainable